cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -205,7 +205,7 @@ func (s *CMServer) Run(_ []string) error {
 	// file, and then overriding the Master flag, if non-empty.
 	kubeconfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
 		&clientcmd.ClientConfigLoadingRules{ExplicitPath: s.Kubeconfig},
-		&clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: s.Master}}).ClientConfig()
+		&clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: s.Master}}).ClientConfig("")
```

Comment:
So here the we will get a config for the legacy v1 group. client.New() will first create a RESTClient with this config, and then create an ExperimentalClient with many hard-coded default values (https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/experimental.go#L126).

I'm not clear what we should do with client.New(). We have these options (or more): 1. ask the user of this function to instead use a versioned client (e.g. pkg/client/v1/) 2. let client.New take a list of config for each group; 3. let client.New() take an unversioned conifg, and use NegotiateVersion to negotiate the version for each group.